### ISSUE TITLE
Multiple issues with newrelic_nrql_alert_condition

### ISSUE THREAD

**Issue by outime** [newrelic/terraform-provider-newrelic#739]
### Terraform Version

+ Terraform v0.12.28
+ provider.newrelic v2.1.1

### Affected Resource(s)

- newrelic_nrql_alert_condition

### Terraform Configuration Files
```hcl
provider "newrelic" {
  account_id = local.newrelic_account_id
  api_key = var.newrelic_api_key
  admin_api_key = var.newrelic_admin_api_key
  region = "US"
}


resource "newrelic_nrql_alert_condition" "replicaset_pods" {
  account_id           = local.newrelic_account_id
  policy_id            = local.newrelic_alert_policy_id
  type                 = "static"
  name                 = "ReplicaSet doesn't have desired amount of pods"
  runbook_url          = "https://example.com/"
  enabled              = true
  value_function       = "single_value"
  violation_time_limit = "TWENTY_FOUR_HOURS"

  nrql {
    query             = "SELECT (podsDesired - podsReady) FROM K8sReplicasetSample where (deploymentName = 'myDeployment')"
    evaluation_offset = 3
  }

  critical {
    operator              = "equal"
    threshold_duration    = 120
    threshold             = 0
    threshold_occurrences = "all"
  }
}
```

### Panic Output

No panic output.

### Expected Behavior

A new NR alert condition is created/updated with the values provided in TF.

### Actual Behavior

No alert condition is created:

```
newrelic_nrql_alert_condition.replicaset_pods: Creating...

Error: Argument "condition" has invalid value $condition.
In field "terms": Expected type "[AlertsNrqlConditionTermsInput!]!", found [{operator: "EQUAL", priority: "CRITICAL", thresholdDuration: 120, thresholdOccurrences: "ALL"}].
In element #1: Expected type "AlertsNrqlConditionTermsInput", found {operator: "EQUAL", priority: "CRITICAL", thresholdDuration: 120, thresholdOccurrences: "ALL"}.
In field "threshold": Expected type "Float!", found null.
In field "operator": Expected type "AlertsNrqlConditionTermsOperator!", found "EQUAL"., null
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

- I could create this same alert condition by hand (i.e. in the UI) without issues.
- Using `threshold` with value `0.0` as suggested in [here](https://github.com/newrelic/terraform-provider-newrelic/issues/660#issuecomment-642067068) reduced the error output to:

```
newrelic_nrql_alert_condition.replicaset_pods: Creating...

Error: Argument "condition" has invalid value $condition.
In field "terms": Expected type "[AlertsNrqlConditionTermsInput!]!", found [{operator: "EQUAL", priority: "CRITICAL", threshold: 0.1, thresholdDuration: 120, thresholdOccurrences: "ALL"}].
In element #1: Expected type "AlertsNrqlConditionTermsInput", found {operator: "EQUAL", priority: "CRITICAL", threshold: 0.1, thresholdDuration: 120, thresholdOccurrences: "ALL"}.
In field "operator": Expected type "AlertsNrqlConditionTermsOperator!", found "EQUAL"., null
```

### References

- https://github.com/newrelic/terraform-provider-newrelic/issues/660 (different resource but offered a sort of workaround for `threshold`)

**Comment by zlesnr** [comment]
Hi, thank you for the report.  From the manifest and the error there, it looks to me like `threshold` needs to be a float, where `0` is an int.  The `operator` also looks like it should be `equals` in place of `equal`.  If that doesn't resolve the issue, I can dig a little further to figure out what's going on here.

**Comment by outime** [comment]
Hi @zlesnr, in my original message I mentioned that I used a float (`0.0`) but the error is similar. About the `operator`, I believe `equal` is actually the correct one as otherwise I get `Error: expected critical.0.operator to be one of [above below equal], got equals`.

**Comment by zlesnr** [comment]
Thanks @outime.  I'm seeing the same message here after I posted.  That seems like a bug to me in the code for "equal" vs "equals".  I see now about the `0.0` attempt, which gives a message containing `0.1`.  The last error there looks to be the one to solve at this point.  In the mean time if you need to get this working while I chase the bug, you could try using "below" or some such to get you closer to your desires.

**Comment by outime** [comment]
@zlesnr FYI I used `operator=below` and `threshold=1.0` and a different error showed up:

```
newrelic_nrql_alert_condition.replicaset_pods: Creating...

Error: A downstream error occurred., [{"extensions":{"code":"BAD_USER_INPUT"},"message":"Validation Error"}]
```

Now it's a bit more difficult to even guess what could be wrong. But hopefully all this is good info for you in order to troubleshoot it.

**Comment by zlesnr** [comment]
Okay, I saw this last week with this API.  Double check the NRQL there, I don't eyeball parse NRQL well, but the following works for me, which is the same as what I think you are using, but with NRQL that I've verified.
```
resource "newrelic_nrql_alert_condition" "replicaset_pods" {
  policy_id            = newrelic_alert_policy.z.id
  type                 = "static"
  name                 = "ReplicaSet doesn't have desired amount of pods"
  runbook_url          = "https://example.com/"
  enabled              = true
  value_function       = "single_value"
  violation_time_limit = "TWENTY_FOUR_HOURS"

  nrql {
    query             = "SELECT percentile(duration, 95) FROM Transaction WHERE appName = 'Dummy App'"
    evaluation_offset = 3
  }

  critical {
    operator              = "below"
    threshold_duration    = 120
    threshold             = 0.1
    threshold_occurrences = "ALL"
  }
}
```

I think once you moved away from using the bad `equal` term, then we moved on to the next error.  If that doesn't end up being what causes the error, let's dig in.  I've opened https://github.com/newrelic/terraform-provider-newrelic/pull/741 for the "equal" naming, but I'll need to release the client first.

**Comment by outime** [comment]
@zlesnr I rewrote the query like `from K8sReplicasetSample SELECT uniqueCount(podsReady) WHERE (deploymentName = 'x' AND clusterName = 'y' AND podsReady = 1)` which narrows down a bit and it worked, thank you!

**Comment by outime** [comment]
@zlesnr one additional issue, this time with `newrelic_infra_alert_condition` (keeping it here as it can be related, otherwise let me know). When you create the alerts by hand, you can see them like this:

![By hand](https://user-images.githubusercontent.com/62993/86229469-6dd01d00-bb98-11ea-9378-eff9db571e7b.png)

When you create the same alert through Terraform, it looks like this:

![Through Terraform](https://user-images.githubusercontent.com/62993/86229526-82acb080-bb98-11ea-80e4-5d0835dd25bc.png)

Functionally they're the same and they both work, but the first one shows the nice narrow down entities, which is very important when you list the alerts by policy as it shows the narrowed down entities in the first case but not in the second, which is quite important info. Do you know how can this be replicated with TF?

**Comment by zlesnr** [comment]
We'd need to compare the response bodies of the API for those two conditions.  I suspect its doable, but I think most of the alerts work is headed towards NRQL alerts instead of other resources like the infra_alert_condition.  Would you mind opening a separate issue about that one since its a different resource?

**Comment by ctrombley** [comment]
@outime I'd like to echo @zlesnr's comment about standardizing on [NRQL conditions](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/nrql_alert_condition).   The guidance from the Alerts team at New Relic is to migrate to NRQL conditions across the board moving forward.  Creating Infra conditions via the Infra API has always had the shortcoming you've outlined here and it's unlikely to be reconciled given the push to move everything over to NRQL conditions.  Please give it a try and let us know if you need any assistance crafting the NRQL query needed to re-implement this specific alert condition.

**Comment by zlesnr** [comment]
I believe the two issues raised here for the `newrelic_nrql_alert_condition` have been resolved in `v2.2.0`.

**Comment by trunet** [comment]
> I believe the two issues raised here for the `newrelic_nrql_alert_condition` have been resolved in `v2.2.0`.

I'm on 2.2.1, still have this problem

**Comment by zlesnr** [comment]
It looks like https://github.com/newrelic/terraform-provider-newrelic/pull/775 should help here also.  Its the zero value you are still having issues with, correct @trunet?

**Comment by trunet** [comment]
I guess so. PR #775 was closed as duplicated of PR #769

**Comment by zlesnr** [comment]
This should be resolved in the next release.  Please let me know if any issues here remain.
